sphinx/builders/linkcheck.py

```diff
@@ -554,6 +561,7 @@ def setup(app: Sphinx) -> Dict[str, Any]:
     # Anchors starting with ! are ignored since they are
     # commonly used for dynamic pages
     app.add_config_value('linkcheck_anchors_ignore', ["^!"], None)
+    app.add_config_value('linkcheck_documents_ignore', [], None)
```

Comment:
The anchor is a part of the reference. So `linkcheck_anchors_ignore` controls how linkcheck builder verifies the reference. In other words, it modifies the target URL itself before checking.

On the other hand, this configuration will control what references should be checked. They are different behavior with `linkcheck_anchors_ignore`. So it would be better to use other words for this configuration, I think.

How about `linkcheck_exclude_documents` instead?